src/sqlfluff/utils/reflow/helpers.py

```diff
@@ -25,3 +31,21 @@ def pretty_segment_name(segment: BaseSegment) -> str:
         # Reference other segments just by their type.
         # (With underscores as spaces)
         return segment.get_type().replace("_", " ")
+
+
+def deduce_line_indent(raw_segment: RawSegment, root_segment: BaseSegment) -> str:
+    """Given a raw segment, deduce the indent of it's line."""
+    seg_idx = root_segment.raw_segments.index(raw_segment)
```

Comment:
This is actually the root segment rather than the parent. The root segment gets passed around for doing exactly this kind of operation.

Reply:
Oh, the `.index()` call confused me (made me think `raw_segment` must be a direct child of `root_segment`, but I guess `.raw_segments` recursively returns **all** raw segments? It's a bit subtle. Should we consider renaming `.raw_segments` as `.raw_descendants` or `.raw_segment_descendants`?

Reply:
Potentially yes - although that turns into a much much bigger fix. I'm going to leave it for now, but I think you're right that it could do with some clarity improvements.